pkg/pgclient/metrics.go

```diff
@@ -5,73 +5,65 @@
 	)
-	statementCacheEnabled = prometheus.NewGauge(
+	writerActive := prometheus.NewGaugeFunc(
 		prometheus.GaugeOpts{
 			Namespace:   util.PromNamespace,
-			Subsystem:   "cache",
-			Name:        "enabled",
-			Help:        "Cache is enabled or not.",
-			ConstLabels: map[string]string{"type": "metric", "name": "statement_cache"},
+			Subsystem:   "sql_database",
+			Name:        "active_connections",
+			Help:        "Number of connections currently active in the pool.",
+			ConstLabels: map[string]string{"pool": "writer"},
+		}, func() float64 {
+			if writerPool == nil {
+				// readonly mode.
+				return 0
+			}
+			return float64(writerPool.Stat().TotalConns())
 		},
 	)
-)
-
-func InitClientMetrics(client *Client) {
-	if statementCacheLen != nil {
-		// Register metrics only once.
-		return
-	}
-	statementCacheLen = createStatementCacheLengthHistogramMetric(client)
-	prometheus.MustRegister(
-		statementCacheEnabled,
-		statementCacheCap,
-		statementCacheLen,
+	readerAcquired := prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace:   util.PromNamespace,
+			Subsystem:   "sql_database",
+			Name:        "acquired_connections",
+			Help:        "Number of connections currently acquired from the pool.",
+			ConstLabels: map[string]string{"pool": "reader"},
+		}, func() float64 {
+			return float64(readerPool.Stat().AcquiredConns())
+		},
 	)
-}
-
-func createStatementCacheLengthHistogramMetric(client *Client) prometheus.Histogram {
-	// we know the upper bound of the cache, so we want
-	// to make that the last bucket of the histogram
-	statementCacheUpperBound := client.metricCache.Cap() * 2
-	// we want to increase the buckets by a factor of 2
-	histogramBucketFactor := 2.0
-	// we want 10 total buckets
-	totalBuckets := 10
-	// If we take the last bucket of the histogram
-	// to be 2 to the power of some x, then
-	// 2^maxFactor=statementCacheUpperBound -> log_2(statementCacheUpperBound) = maxFactor
-	maxFactor := math.Floor(math.Log2(float64(statementCacheUpperBound)))
-	// Each bucket is calculated as 2^x, x being incremented by 1 for each bucket
-	// ending with 2^maxFactor. To find the start bucket of the histogram,
-	// We need to find minFactor so we can end up with 10 buckets
-	minFactor := maxFactor - float64(totalBuckets)
-	minFactor = math.Max(minFactor, 1) // in case maxFactor <= 10
-
-	histogramStartBucket := math.Pow(histogramBucketFactor, minFactor)
-	return prometheus.NewHistogram(
-		prometheus.HistogramOpts{
+	readerActive := prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
 			Namespace:   util.PromNamespace,
-			Subsystem:   "cache",
-			Name:        "elements_histogram",
-			Help:        "Number of elements in cache in terms of elements count.",
-			Buckets:     prometheus.ExponentialBuckets(histogramStartBucket, histogramBucketFactor, totalBuckets),
-			ConstLabels: map[string]string{"type": "metric", "name": "statement_cache"},
+			Subsystem:   "sql_database",
+			Name:        "active_connections",
```

Comment:
Same here. Maybe better name `total_connections` since it includes Active + Idle